scripts/pre_commit_linter.py

```diff
@@ -505,6 +511,83 @@ def _check_newline_character(all_files):
     return summary_messages
 
 
+def _get_ng_template_id(line):
+    """This function returns the id of the ng-template, if present.
+    The function assumes id is writted as id="id_name" i.e there is no space
+    before and after '=' and id_name is wrapped in '\"'
+
+    Args:
+        line: str. Line to search for ng-template.
+
+    Returns:
+        str or None. Id of the tag or None.
+    """
+    template_pattern = re.compile(".*text\/ng-template")
+    line = template_pattern.match(line)
+
+    if line is not None:
+        line = line.string
+    else:
+        return None
+
+    line = line.split(' ')
+    for attr in line:
+        attr_value = attr.split('=')
+
+        if attr_value[0] == 'id':
+            return attr_value[1].strip().strip('\"\>')
+
+    return None
+
+
+def _check_directives_directly_referenced(all_files):
+    """This function is used to check that the directives which don't use jinja templating
+    are directly referenced in js files.
+    """
+
+    print 'Starting directives-directly-referenced check'
+    print '-----------------------------------------'
+    check_failed = 0
+    file_pattern = re.compile(".*core.*\.html$")
+    all_files = [
+        file_name for file_name in all_files
+        if file_pattern.match(file_name) is not None ]
+    total_files_checked = len(all_files)
+    total_error_count = 0
+
+    for file_name in all_files:
+        f = open(file_name, 'rb')
```

Comment:
'rb' --> 'r' 

Though this doesn't make any difference in linux, I realized that it may create problem in windows.

Reply:
Also, should use `with open(..., ...) as f:`